store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -38,10 +39,44 @@ public class OwaspHtmlSanitizer implements Callable<String> {
     private String vHost;
 
     public OwaspHtmlSanitizer(String html, boolean neuterImages, String vHost) {
-        this.html = html;
+        // fix to check extra tags & remove
+        if (StringUtil.isNullOrEmpty(html)) {
+            this.html = html;
+        } else {
+            this.html = checkUnbalancedTags(html);;
+        }
         this.neuterImages = neuterImages;
         this.vHost = vHost;
     }
+    
+    /** 
+     * A method to check & remove unbalanced tags which may have started, not ended
+     * 
+     * @return the Formatted Html after removing unbalanced tags
+     */
+    private String checkUnbalancedTags(String html) {
+        ArrayList<String> tags = new ArrayList<String>();
```

Comment:
null check for html should be in this method, so if this method is used else where it  won't  lead to NPE

Reply:
added inside the method